### PR TITLE
[SC-1086] Natspecs and README

### PR DESCRIPTION
**Pull request by zZoMROT**

### CODE REVIEW

File: docs/js/.nojekyll [hunk 1]
```diff
@@ -0,0 +1 @@
+TypeDoc added this file to prevent GitHub Pages from using Jekyll. You can turn off this behavior by setting the `githubPages` option to false.
```
Comment (ZumZoom) [inline docs/js/.nojekyll:1]: We should probably do what is suggested here and set the `githubPages` option to false as we do not intend to use GH Pages in any foreseeable future.

File: contracts/PermitAndCall.sol [hunk 2]
```diff
@@ -5,9 +5,19 @@ pragma solidity ^0.8.0;
 import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import { SafeERC20 } from "./libraries/SafeERC20.sol";
 
+/**
+ * @title PermitAndCall
+ * @dev Abstract contract to support permit and action execution in a single transaction.
+ * Allows tokens that implement EIP-2612 permits to be approved and spent in a single transaction.
```
Comment (ZumZoom) [inline contracts/PermitAndCall.sol:11]: Not only EIP-2612 but also DAI-like permits and Permi2.

File: contracts/PermitAndCall.sol [hunk 3]
```diff
@@ -5,9 +5,19 @@ pragma solidity ^0.8.0;
...
+ */
 abstract contract PermitAndCall {
     using SafeERC20 for IERC20;
 
+    /**
+     * @notice Executes a permit for an ERC20 token and then a specified action in a single transaction.
+     * @param permit The permit data for the ERC20 token, allowing this contract to spend the token.
```
Comment (ZumZoom) [inline contracts/PermitAndCall.sol:18]: We should also describe the format of `permit` data. At least mention that the first 20 bytes are token address.

File: contracts/interfaces/ICreate3Deployer.sol [hunk 4]
```diff
@@ -2,7 +2,23 @@
 
 pragma solidity ^0.8.0;
 
+/**
+ * @title ICreate3Deployer
+ * @dev Interface for deploying contracts with deterministic addresses via CREATE2.
```
Comment (ZumZoom) [inline contracts/interfaces/ICreate3Deployer.sol:7]:
```suggestion
 * @dev Interface for deploying contracts with deterministic addresses via CREATE3.
```

**Review by SevenSwen — APPROVED** [review]

**Review by ZumZoom — APPROVED** [review]

**Pull request merged**